src/FluentMigrator.Runner/MigrationRunner.cs

```diff
@@ -280,7 +259,6 @@ public virtual void ApplyMigrationUp(IMigrationInfo migrationInfo, bool useTrans
                 _alreadyOutputPreviewOnlyModeWarning = true;
             }
 
-            if (!migrationInfo.IsAttributed() || !VersionLoader.VersionInfo.HasAppliedMigration(migrationInfo.Version))
```

Comment:
The logic indicated in that `if` is duplicated by local callers to the function; that is, for every invocation within the project, it evaluates to "true".  There were no independent tests of that function, but being public it could be consumed elsewhere.